tests/components/modbus/conftest.py

```diff
@@ -37,61 +40,84 @@ def __init__(self, register_words):
         self.bits = register_words
 
 
-async def setup_base_test(
+async def base_test(
     sensor_name,
     hass,
-    use_mock_hub,
     data_array,
     entity_domain,
     scan_interval,
+    register_words,
+    expected,
+    method_discovery=False,
 ):
-    """Run setup device for given config."""
+    """Run test on device for given config."""
 
     # Full sensor configuration
-    config = {
-        entity_domain: {
-            CONF_PLATFORM: "modbus",
-            CONF_SCAN_INTERVAL: scan_interval,
-            **data_array,
+    if method_discovery:
+        config = {
+            DOMAIN: {
+                CONF_NAME: DEFAULT_HUB,
+                CONF_TYPE: "tcp",
+                CONF_HOST: "modbusTest",
+                CONF_PORT: 5001,
+                **data_array,
+            },
+        }
+    else:
+        config = {
+            DOMAIN: {
+                CONF_NAME: DEFAULT_HUB,
+                CONF_TYPE: "tcp",
+                CONF_HOST: "modbusTest",
+                CONF_PORT: 5001,
+            },
+        }
+        configDeviceOLD = {
+            entity_domain: {
+                CONF_PLATFORM: DOMAIN,
+                CONF_SCAN_INTERVAL: scan_interval,
+                **data_array,
+            }
         }
-    }
 
-    # Initialize sensor
+    # mock timer and add modbus platform with devices (new config)
+    # first add modbus platform, then devices (old config)
     now = dt_util.utcnow()
     with mock.patch("homeassistant.helpers.event.dt_util.utcnow", return_value=now):
-        assert await async_setup_component(hass, entity_domain, config)
+        # setup modbus platform
+        if method_discovery:
+            await async_load_platform(
```

Comment:
Setting up the integration should also set up the platforms. Why do we need to do that in the test specifically?